source/docs/essentials/add-new-attribute.md

```diff
@@ -0,0 +1,156 @@
+
+```js
+// mymodule/server/modules/acme/schema.gql
+extend type Product {
+  rate: String
+}
+```
+
+### Implement the resolver
+
+We must now define what Front-Commerce should fetch when `rate` is requested in a GraphQL query ; in other words we must write the code that will **resolve** the queries for the rates.
+
+Create a file named `resolvers.js` and type in the following code. It implements the resolver.
+
+```js
+// mymodule/server/modules/acme/resolvers.js
+export default {
+  Product: {
+    rate: parent => {
+      return parent.rate;
+    }
+  }
+};
+```
+
+### Index our changes
+
+If we want our code to be taken into account when the module is loaded, we must index it.
+
+Create a file named `index.js` and type in the following code. It indexes the schema and the resolver from earlier.
+
+```js
+// mymodule/server/modules/acme/index.js
+import typeDefs from "./schema.gql";
+import resolvers from "./resolvers";
+
+export default {
+  namespace: "Acme",
+  typeDefs: typeDefs,
+  resolvers: resolvers
+};
+```
+
+### Overriding the files
+
+In **this** example, the files to be overridden are `ProductSynthesisFragment.gql` and `Synthesis.js` both located at `node_modules/front-commerce/src/web/theme/modules/ProductView/Synthesis/`. Copy and paste them into your folder.
+
+### Making the query
+
+In `ProductSynthesisFragment;gql`, add the following query line. It means that the application will request the `rate` field as well when sending the query. (Depending on the version of Front-Commerce you are using, the content might differ slightly.)
+
+```diff
+#import "theme/components/organisms/Configurator/ProductConfiguratorFragment.gql"
+#import "theme/modules/AddToCart/ProductStockFragment.gql"
+#import "theme/components/atoms/Typography/Price/ProductPriceFragment.gql"
+#import "theme/modules/ProductView/Details/ProductDetailsFragment.gql"
+
+fragment ProductSynthesisFragment on Product {
+  sku
+  name
+  ...ProductPriceFragment
+  description
++rate
+  ...ProductConfiguratorFragment
+  ...ProductStockFragment
+  ...ProductDetailsFragment
+  configurations {
+    product {
+      sku
+      ...ProductPriceFragment
+    }
+  }
+}
+```
+
+### Displaying the result on screen
+
+We're almost there! In the file `Synthesis.js` (which generates the product page), you can add some code to display the `rate`. Knowing React.js or at least HTML syntax is necessary to perform this step without screwing the other items on the page.
+
+```js
+// Many more things here
+const ProductSynthesis = props => {
+  const {
+    product
+    // Many more things here
+  } = props;
+  return (
+    <div className="product__synthesis">
+      {/* Many more things here */}
+      <p>Rate: {product.rate} / 5</p>
+      {/* Many more things here */}
+    </div>
+  );
+};
+
+export default ProductSynthesis;
+```
+
+## Add the module
+
+Last step: adding the module we just made to Front-Commerce. Edit the `.front-commerce.js` file (root folder) as shown below to do so:
```

Comment:
Ah, I've just read this part. I feel like it's a bit too late. People will want to test things out before having the full things wired up. It would be great to tell people along the way how to test that everything is working (graphql playground, console.log, etc.) and tell them when they need to restart the server.